vertx-service-discovery-backend-redis/src/main/java/io/vertx/servicediscovery/backend/redis/RedisBackendService.java

```diff
@@ -21,30 +21,86 @@
 import io.vertx.core.Handler;
 import io.vertx.core.Vertx;
 import io.vertx.core.json.JsonObject;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.redis.client.*;
 import io.vertx.servicediscovery.Record;
 import io.vertx.servicediscovery.spi.ServiceDiscoveryBackend;
-import io.vertx.redis.RedisClient;
-import io.vertx.redis.RedisOptions;
 
 import java.util.List;
 import java.util.Objects;
 import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
 
+import static io.vertx.redis.client.Request.cmd;
+import static io.vertx.redis.client.Command.*;
+
 /**
  * An implementation of the discovery backend based on Redis.
  *
  * @author <a href="http://escoffier.me">Clement Escoffier</a>
  */
 public class RedisBackendService implements ServiceDiscoveryBackend {
 
-  private RedisClient redis;
+  private Redis redis;
   private String key;
 
+  private static final int DISCONNECTED = 0;
+  private static final int CONNECTING = 1;
+  private static final int CONNECTED = 2;
+
+  private final AtomicInteger state = new AtomicInteger();
+
   @Override
   public void init(Vertx vertx, JsonObject configuration) {
     key = configuration.getString("key", "records");
-    redis = RedisClient.create(vertx, new RedisOptions(configuration));
+
+    System.out.println(configuration.encodePrettily());
+
+    if (configuration.containsKey("host") || configuration.containsKey("port")) {
+      redis = Redis.createClient(vertx, new RedisOptions(configuration).setEndpoint(
+        SocketAddress.inetSocketAddress(
+          configuration.getInteger("port", 6379),
+          configuration.getString("host", "localhost"))
+      ));
+    } else {
+      redis = Redis.createClient(vertx, new RedisOptions(configuration));
+    }
+  }
+
+  private void redisCall(Request request, Handler<AsyncResult<Response>> handler) {
```

Comment:
Take a look at the Vert.x Cassandra Client and how the session is managed. It uses execute blocking and dedicated `TaskQueue`. It allows to manage the connection state without failing requests if the client is connecting.